agave_app/renderer.cpp

```diff
@@ -242,6 +273,12 @@ Renderer::processRequest()
     // QMetaObject::invokeMethod(
     //  renderDialog, [=]() { /* ... onRenderRequestProcessed(lastReq, img); ... */ }, Qt::QueuedConnection);
     emit requestProcessed(lastReq, img);
+
+    if (m_streamMode && !shouldContinue()) {
+      m_frameIterations = 0;
+      m_frameTimeSeconds = 0.0f;
+      emit frameDone(img);
+    }
```

Comment:
this is the check that sends a new message back to the main thread that a rendered frame was completed, so the main thread can update UI progress bars and save the image to disk.